addons/mrp_subcontracting_dropshipping/models/stock_warehouse.py

```diff
@@ -15,6 +15,54 @@ class StockWarehouse(models.Model):
         'stock.rule', 'Subcontracting-Dropshipping MTS Rule'
     )
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        # if new warehouse has resupply enabled, enable global route
+        if any(['subcontracting_dropshipping_to_resupply' in vals for vals in vals_list]):
+            res.update_global_route_dropship_subcontractor()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        # if all warehouses have resupply disabled, disable global route, until its enabled on a warehouse
+        if 'subcontracting_dropshipping_to_resupply' in vals or 'active' in vals:
+            self._update_dropship_subcontract_rules()
+            self.update_global_route_dropship_subcontractor()
```

Comment:
Be careful, you might only write to some warehouses, so the `self.company_id.dropship_subcontractor_pick_type_id.active =` lines in `update_global_route_dropship_subcontractor` will only apply to the warehouses you are writing to. I think you will have to make `update_global_route_dropship_subcontractor` more generic to apply to all warehouses.